hack/e2e-test.sh

```diff
@@ -236,22 +262,36 @@ INSTALL_RESULT=""
 
 i=1
 while [ $i -le ${max_cluster_deployment_status_checks} ]; do
-  IS_CLUSTER_DEPLOYMENT_INSTALLED=$(oc get cd ${CLUSTER_NAME} -n ${CLUSTER_NAMESPACE} -o json | jq .spec.installed )
-  if [[ "${IS_CLUSTER_DEPLOYMENT_INSTALLED}" == "true" ]] ; then
+  CD_JSON=$(oc get cd ${CLUSTER_NAME} -n ${CLUSTER_NAMESPACE} -o json)
+  if [[ $(jq .spec.installed <<<"${CD_JSON}") == "true" ]] ; then
     INSTALL_RESULT="success"
     break
   fi
+  PF_COND=$(jq -r '.status.conditions[] | select(.type == "ProvisionFailed")' <<<"${CD_JSON}")
+  if [[ $(jq -r .status <<<"${PF_COND}") == 'True' ]]; then
```

Comment:
nts: This works when `$PF_COND` is empty or contains no `.status`. Specifically:

```bash
if [[ $something_empty == 'True' ]]
```
works, because it evaluates to
```bash
if [[ '' == \T\r\u\e ]]
```
and not
```bash
if [[ == \T\r\u\e ]]
```